resource-managers/yarn/pom.xml

```diff
@@ -88,6 +88,13 @@
       <artifactId>hadoop-client</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>jakarta.servlet</groupId>
+      <artifactId>jakarta.servlet-api</artifactId>
+      <version>4.0.3</version>
+      <scope>test</scope>
+    </dependency>
+
```

Comment:
cc  @srowen I found we add `org.glassfish.jersey.test-framework.providers` dependency after https://github.com/apache/spark/pull/25455/files

But I'm not sure should we use `jakarta.servlet-api` as a compile dependency  instead of `javax.servlet-api` directly

Reply:
How about instead excluding the 4.0.x dependency from wherever it comes in? My guess is we want the 3.x version.

Reply:
And I'm not sure should we need add this test dependency, but it failed in my test environment without this dependency

Reply:
I've tried it once, but failed. It looks like really need 4.0.x, but I'll re-check this.

Reply:
OK, then this approach seems OK. We are in jar hell but what else is new!

Reply:
Try to excluding 4.0.x from parent pom.xml and `core/pom.xml` as follow:

![image](https://user-images.githubusercontent.com/1475305/93795937-b580b500-fc6c-11ea-8bf0-2e5a9f7a4e48.png)

then and remove `jakarta.servlet-api` dependency from `resource-managers/yarn/pom.xml` and rebuild&rerun the test case but still failed :( ..., It seems that 4.0.x is needed in the test

Reply:
Fortunately, there are no real failed test cases of yarn module in Scala 2.13 :)